File: StringToBinary/StringToBinary.py


#This function takes character and converts it to deciaml ascii
def charToAscii(char):
    return(ord(char))

#Takesa a string and converts it to a list of decimal ascii
def stringToAscii(string):
    decimalNumList = []
    for char in string:
        decimalNumList.append(charToAscii(char))
    return decimalNumList

#Takes a list of decimal ascii and converts it to binary string
def asciiToBinary(decimalNumList):
    binaryString = ""
    for num in decimalNumList:
        start = 64
        binaryNum = ""
        for binaryPlace in range(0,7):
            if(num >= start):
                num = num - start
                binaryNum += '1'
            else:
                binaryNum += '0'
            start = start/2
        binaryString += binaryNum + " "
    binaryString = binaryString[0:len(binaryString)-1]
    return binaryString

#takes a string and converts it to string binary
def stringToBinary(string):
    asciiList = stringToAscii(string)
    binary = asciiToBinary(asciiList)
    return binary

myBinary = stringToBinary("Hello This is my test String")
print(myBinary)


def pythonBinary(string):
    return (' '.join(format(ord(x), 'b') for x in string))



pythBinary = pythonBinary("Hello This is my test String")
print(pythBinary)


print(type(pythBinary))

File: StringToBinary/StringToBinary.js

//Takes character and converts it to deciaml ascii
function charToAscii(ch){
    return ch.charCodeAt(0);
}

//Takes a string and converts it to a list of decimal ascii
function stringToAscii(str){

    let splitString = str.split("")
    const decimalList = splitString.map(ch => charToAscii(ch))
    return decimalList
}

//Takes a list of decimal ascii and converts it to binary string
function asciiToBinary(decimalList){
    var binaryString = ""

    const binaryList = decimalList.map(num => {
        let start = 64
        let binary = ""
        for(let i = 0; i < 7; i++){
             if(num >= start){
                num = num - start
                binary += '1'
             }else{
                binary += '0'
             }
            start = start/2
        }

        return binary
    })

    return binaryList.join(" ")
}

//takes a string and converts it to string binary
function stringToBinary(str){
    var asciiList = stringToAscii(str)
    var binaryStr = asciiToBinary(asciiList)
    return binaryStr
}

let mStr = "HELLO THIS is MYT"

console.log(stringToBinary(mStr))
